graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java

```diff
@@ -738,10 +737,14 @@ private void createLookupCaches(String bundleId, Set<LookupCacheBundle> lookupCa
         final CountDownLatch latch = new CountDownLatch(caches.size());
         caches.forEach(c -> c.addListener(new LatchUpdaterListener(latch), scheduler));
 
-        latch.await(30, TimeUnit.SECONDS);
+        try {
+            latch.await(30, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            LOG.warn("Starting imported Lookup Table Caches did not finish within 30 seconds. A server restart might be required for imported Lookup Tables to function.");
```

Comment:
The `InterruptedException` is not thrown when the latch's timeout expires but only when the thread is being interrupted.

If the timeout expired, [`CountDownLatch#await()`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CountDownLatch.html#await-long-java.util.concurrent.TimeUnit-) will return `false`.

Reply:
✅ (Should read javadocs properly)